addons/auth_signup/models/res_users.py

```diff
@@ -200,7 +200,13 @@ def action_reset_password(self):
             # TDE FIXME: make this template technical (qweb)
             with self.env.cr.savepoint():
                 force_send = not(self.env.context.get('import_file', False))
-                template.send_mail(user.id, force_send=force_send, raise_exception=True, email_values=email_values)
+                try:
+                    template.send_mail(user.id, force_send=force_send, raise_exception=True, email_values=email_values)
+                except MailDeliveryException as mde:
+                    if len(mde.args) == 2 and isinstance(mde.args[1], ConnectionRefusedError):
+                        raise UserError(_("Could not contact the mail server, please check your outgoing email server configuration"))
+                    else:
+                        raise UserError(_("There was an error when trying to deliver your Email, please check your configuration"))
```

Comment:
In `Create` of this class, make sure to catch UserError now instead of MailDeliveryException.

Reply:
Hello @thiryr 
changes have been done, kindly review them.
Thanks.